### PR TITLE
Invoice maker

### PR DESCRIPTION
**Pull request by cvera-tech**
Client actions and WorkType actions complete. Started WorkDone actions.

### CODE REVIEW

**Comment by jbvera** [comment]
End of week.
- Completed refactoring to EF.
- Completed Invoices Index, Details, and Edit actions.
- Started Invoices AddLineItem action.

File: InvoiceMaker/InvoiceMaker/Views/Clients/Create.cshtml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿@model InvoiceMaker.FormModels.CreateClient
+
+<h1>Create a Client</h1>
+
+@using (Html.BeginForm())
+{
+    @Html.AntiForgeryToken()
```
Comment (appacademy-jchurchill) [inline InvoiceMaker/InvoiceMaker/Views/Clients/Create.cshtml]: Don't forget to add the `ValidateAntiForgeryToken` attribute to the POST action method.

File: InvoiceMaker/Seed Data.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+* Window Washing, 15.75
+* Window Soiling, 22.50
+* Hang Gliding, 53.40
+* Kessel Run, 11.99
+* Ritual Chanting, 42.54
+
+# Works Done
+* Jar jar Binkle, Window Soiling, 9/20/2019 2:00 PM
+* Jar jar Binkle, Window Washing, 9/21/2019 4:45 PM
+* Mickey Mouse, Ritual Chanting, 12/31/2019 11:45 PM
```
Comment (appacademy-jchurchill) [inline InvoiceMaker/Seed Data.md]: I had no idea that Mickey Mouse was into ritual chanting. Good to know :)

**Review by appacademy-jchurchill — APPROVED** [review]
Nice job making progress on adding support for invoices! Your code continues to be clean and easy to read and understand.

File: InvoiceMaker/InvoiceMaker/Data/InvoiceInitializer.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                new WorkType() { Name = "Window Washing", Rate = 15.75m },
+                new WorkType() { Name = "Window Soiling", Rate = 22.50m },
+                new WorkType() { Name = "Hang Gliding", Rate = 53.40m },
+                new WorkType() { Name = "Kessel Run", Rate = 11.99m },
+                new WorkType() { Name = "Ritual Chanting", Rate = 42.54m }
+            };
+
+            // Navigation property must be set, not just the foreign key!
+            // Otherwise EF cannot determine the principal entity in the relationship
```
[the remark this answers is not in the document]
Comment (appacademy-jchurchill) [reply InvoiceMaker/InvoiceMaker/Data/InvoiceInitializer.cs]: I wonder if it'd make a difference if you defined both sides of the relationship within your entity models (i.e. foreign key and navigation property on the dependent side and a collection property on the principal side).

**Pull request merged**